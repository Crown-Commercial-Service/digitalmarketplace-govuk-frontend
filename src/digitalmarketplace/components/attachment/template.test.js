/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('attachment')

describe('Attachment', () => {
  describe('by default', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples.default)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('attachment', examples.default, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('renders a thumbnail', async () => {
      const $ = render('attachment', examples.default)
      const thumbnail = $('.dm-attachment__thumbnail--doc')

      expect(thumbnail.length).toBe(1)
    })

    it('renders a download link with correct text and url', async () => {
      const $ = render('attachment', examples.default)
      const link = $('.dm-attachment__link')

      expect(link.text().trim()).toEqual('Default attachment')
      expect(link.attr('href')).toEqual('https://digitalmarketplace.service.gov.uk/attachment')
    })

    it('does not render metadata', async () => {
      const $ = render('attachment', examples.default)
      const metadata = $('.dm-attachment__metadata')

      expect(metadata.length).toBe(0)
    })
  })

  describe('if spreadsheet', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples.spreadsheet)
      expect($.html()).toMatchSnapshot()
    })

    it('renders the correct thumbnail', async () => {
      const $ = render('attachment', examples.spreadsheet)
      const thumbnail = $('.dm-attachment__thumbnail--csv')

      expect(thumbnail.length).toBe(1)
    })

    it('renders the correct filetype and abbreviation', async () => {
      const $ = render('attachment', examples.spreadsheet)
      const filetype = $('.dm-attachment__attribute')
      const abbr = $('.dm-attachment__abbr')

      expect(filetype.text().trim()).toEqual('CSV')
      expect(abbr.attr('title')).toEqual('Comma-separated Values')
    })
  })

  describe('if pdf', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples.pdf)
      expect($.html()).toMatchSnapshot()
    })

    it('renders the correct thumbnail', async () => {
      const $ = render('attachment', examples.pdf)
      const thumbnail = $('.dm-attachment__thumbnail--pdf')

      expect(thumbnail.length).toBe(1)
    })

    it('renders the correct filetype', async () => {
      const $ = render('attachment', examples.pdf)
      const filetype = $('.dm-attachment__attribute')
      const abbr = $('.dm-attachment__abbr')

      expect(filetype.text().trim()).toEqual('PDF')
      expect(abbr.attr('title')).toEqual('Portable Document Format')
    })
  })

  describe('if unknown filetype', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['unknown filetype'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders a generic thumbnail', async () => {
      const $ = render('attachment', examples['unknown filetype'])
      const thumbnail = $('.dm-attachment__thumbnail--doc')

      expect(thumbnail.length).toBe(1)
    })

    it('renders the raw filetype', async () => {
      const $ = render('attachment', examples['unknown filetype'])
      const filetype = $('.dm-attachment__attribute')

      expect(filetype.text().trim()).toEqual('txt')
    })
  })

  describe('with file size', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['with file size'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders the file size', async () => {
      const $ = render('attachment', examples['with file size'])
      const attributes = $('.dm-attachment__attribute')
      expect(attributes.length).toBe(2)
      expect($(attributes[1]).text().trim()).toEqual('21.5 KB')
    })
  })

  describe('with page count', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['with page count'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders the page count', async () => {
      const $ = render('attachment', examples['with page count'])
      const attributes = $('.dm-attachment__attribute')
      expect(attributes.length).toBe(2)
      expect($(attributes[1]).text().trim()).toEqual('12 pages')
    })
  })

  describe('with alternative format contact email', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['with alternative format contact email'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders a details component', async () => {
      const $ = render('attachment', examples['with alternative format contact email'])
      const details = $('.govuk-details')
      expect(details.length).toBe(1)

      const summary = $('.govuk-details__summary')
      expect(summary.text().trim()).toEqual('Request an accessible format')

      const detailText = $('.govuk-details__text')
      expect(detailText.text()).toContain(examples['with alternative format contact email'].alternativeFormatContactEmail)
    })
  })

  describe('with alternative format HTML', () => {
    it('matches existing snapshot', () => {
      const $ = render('option-select', examples['with alternative format HTML'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders a details component', async () => {
      const $ = render('attachment', examples['with alternative format HTML'])
      const details = $('.govuk-details')
      expect(details.length).toBe(1)

      const summary = $('.govuk-details__summary')
      expect(summary.text().trim()).toEqual('Request an accessible format')

      const detailText = $('.govuk-details__text')
      expect(detailText.html().trim()).toEqual(examples['with alternative format HTML'].alternativeFormatHtml)
    })
  })

  describe('as text only', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['as only text'])
      expect($.html()).toMatchSnapshot()
    })

    it('does not render thumbnail', async () => {
      const $ = render('attachment', examples['as only text'])
      const thumbnail = $('.dm-attachment__thumbnail')

      expect(thumbnail.length).toBe(0)
    })

    it('renders class attribute', async () => {
      const $ = render('attachment', examples['as only text'])
      const link = $('.' + examples['as only text'].link.classes)

      expect(link.length).toBe(1)
    })

    it('parenthesizes attributes', async () => {
      const $ = render('attachment', examples['as only text'])
      const linkText = $('.dm-attachment-link').text().trim()

      expect(linkText).toContain('Attachment with no thumbnail')
      expect(linkText).toContain('(PDF, 21.5 KB)')
    })
  })

  describe('with description', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['with description'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders a description', async () => {
      const $ = render('attachment', examples['with description'])
      const description = $('.govuk-hint')

      expect(description.text().trim()).toEqual('This is the agreement you signed.')
    })
  })

  describe('with custom heading level', () => {
    it('matches existing snapshot', () => {
      const $ = render('attachment', examples['with custom heading level'])
      expect($.html()).toMatchSnapshot()
    })

    it('renders a description', async () => {
      const $ = render('attachment', examples['with custom heading level'])
      const heading = $('h3')

      expect(heading.text().trim()).toEqual('Attachment with custom heading level')
    })
  })
})
