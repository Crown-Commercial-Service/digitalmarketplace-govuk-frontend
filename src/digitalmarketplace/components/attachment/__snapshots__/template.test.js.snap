// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment as text only matches existing snapshot 1`] = `
"<html><head></head><body><span class=\\"dm-attachment-link govuk-body\\">
    <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with no thumbnail</a>
    (<abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr>, 21.5 KB)
  </span>
</body></html>"
`;

exports[`Attachment by default matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--doc\\" version=\\"1.1\\" viewBox=\\"0 0 84 120\\" width=\\"84\\" height=\\"120\\" aria-hidden=\\"true\\">
            <path d=\\"M74.85 5v106H5\\" fill=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
            <path d=\\"M79.85 10v106H10\\" fill=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Default attachment</a>
      </h2>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment if pdf matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Portable Document Format attachment</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment if spreadsheet matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.csv\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--csv\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zm0 47h18.75v63H12zm55 2v59H51V61h16m2-2H49v63h20V59z\\" stroke-width=\\"0\\"/>
            <path d=\\"M49 61.05V120H32.8V61.05H49m2-2H30.75v63H51V59zm34 2V120H69.05V61.05H85m2-2H67v63h20V59z\\" stroke-width=\\"0\\"/>
            <path d=\\"M30 68.5h56.5M30 77.34h56.5M30 112.7h56.5M30 95.02h56.5M30 86.18h56.5M30 103.86h56.5\\" fill=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.csv\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Comma-separated values attachment</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Comma-separated Values\\" class=\\"dm-attachment__abbr\\">CSV</abbr></span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment if unknown filetype matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.txt\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--doc\\" version=\\"1.1\\" viewBox=\\"0 0 84 120\\" width=\\"84\\" height=\\"120\\" aria-hidden=\\"true\\">
            <path d=\\"M74.85 5v106H5\\" fill=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
            <path d=\\"M79.85 10v106H10\\" fill=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.txt\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Unknown filetype attachment</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\">txt</span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with alternative format HTML matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-option-select\\" data-module=\\"dm-option-select\\">
    <h2 class=\\"dm-option-select__heading js-container-heading\\">
      <span class=\\"dm-option-select__title js-container-button\\" id=\\"option-select-title-undefined\\"></span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"/></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-option-select__icon dm-option-select__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"/></svg>
    </h2>

    <div role=\\"group\\" aria-labelledby=\\"option-select-title-undefined\\" class=\\"dm-option-select__container js-options-container\\" id tabindex=\\"-1\\">
      <div class=\\"dm-option-select__container-inner js-auto-height-inner\\">
      
<div class=\\"govuk-form-group\\">
<fieldset class=\\"govuk-fieldset\\">
  <div class=\\"govuk-checkboxes govuk-checkboxes--small\\">
  </div>
</fieldset>
</div>

      </div>
    </div>
</div>
</body></html>"
`;

exports[`Attachment with alternative format contact email matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with alternative format contact email</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
        <details class=\\"govuk-details\\">
  <summary class=\\"govuk-details__summary\\">
    <span class=\\"govuk-details__summary-text\\">
      Request an accessible format
    </span>
  </summary>
  <div class=\\"govuk-details__text\\">
        If you use assistive technology (such as a screen reader) and need a version of this document in a more accessible format, please email <a href=\\"mailto:help@digitalmarketplace.service.gov.uk\\" target=\\"_blank\\" class=\\"govuk-link\\">help@digitalmarketplace.service.gov.uk</a>. Please tell us what format you need. It will help us if you say what assistive technology you use.

  </div>
</details>

    </div>
  </section>
</body></html>"
`;

exports[`Attachment with custom heading level matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h3 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with custom heading level</a>
      </h3>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with description matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with description</a>
      </h2>
        <div class=\\"govuk-hint\\">This is the agreement you signed.</div>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with file size matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with file size</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span>, <span class=\\"dm-attachment__attribute\\">21.5 KB</span>
</p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with last updated matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with last updated</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span>, <span class=\\"dm-attachment__attribute\\">Last updated: <time datetime=\\"2020-01-01\\">New year&apos;s day</time></span>
</p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with link attributes matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" key1=\\"value1\\" key2=\\"value2\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title govuk-!-font-size-19\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\" key1=\\"value1\\" key2=\\"value2\\">Attachment with link attributes</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with page count matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with page count</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span>, <span class=\\"dm-attachment__attribute\\">12 pages</span>
</p>
    </div>
  </section>
</body></html>"
`;

exports[`Attachment with small thumbnail matches existing snapshot 1`] = `
"<html><head></head><body><section class=\\"dm-attachment\\" data-module=\\"dm-attachment\\">
    <div class=\\"dm-attachment__thumbnail dm-attachment__thumbnail--small\\">
      <a class=\\"govuk-link\\" href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" target=\\"_self\\" tabindex=\\"-1\\" aria-hidden=\\"true\\">
          <svg class=\\"dm-attachment__thumbnail-image dm-attachment__thumbnail--pdf dm-attachment__thumbnail-image--small\\" version=\\"1.1\\" viewBox=\\"0 0 99 140\\" width=\\"99\\" height=\\"140\\" aria-hidden=\\"true\\">
            <path d=\\"M12 12h75v27H12zM12 59h9v9h-9zM12 77h9v9h-9zM12 95h9v9h-9zM12 113h9v9h-9zM30 59h57v9H30zM30 77h39v9H30zM30 95h57v9H30zM30 113h48v9H30z\\" stroke-width=\\"0\\"/>
          </svg>
      </a>
    </div>
    <div class=\\"dm-attachment__details\\">
      <h2 class=\\"dm-attachment__title govuk-!-font-size-19\\">
        <a href=\\"https://digitalmarketplace.service.gov.uk/attachment.pdf\\" class=\\"govuk-link dm-attachment__link\\" target=\\"_self\\">Attachment with small thumbnail</a>
      </h2>
        <p class=\\"dm-attachment__metadata\\">
          <span class=\\"dm-attachment__attribute\\"><abbr title=\\"Portable Document Format\\" class=\\"dm-attachment__abbr\\">PDF</abbr></span></p>
    </div>
  </section>
</body></html>"
`;
